killerbee/pcapdump.py

```diff
@@ -103,27 +106,24 @@ def pcap_next(self):
 
 
 class PcapDumper:
-    def __init__(self, datalink, savefile, ppi = False):
+    def __init__(self, datalink, savefile, ppi = False, folder='.'):
         '''
         Creates a libpcap file using the specified datalink type.
         @type datalink: Integer
         @param datalink: Datalink type, one of DLT_* defined in pcap-bpf.h
-        @type savefile: String or file-like object
-        @param savefile: Output libpcap filename to open, or file-like object
-        @type ppi: Boolean
-        @param ppi: Include CACE Per-Packet Information (defaults to False)
+        @type savefile: String
+        @param savefile: Output libpcap filename to open
         @rtype: None
         '''
         if ppi: from killerbee.pcapdlt import DLT_PPI
-        self.ppi = ppi
-
-        if isinstance(savefile, basestring):
-            self.__fh = open(savefile, mode='wb')
-        elif hasattr(savefile, 'write'):
-            self.__fh = savefile
-        else:
-            raise ValueError("Unsupported type for 'savefile' argument")
-
+        self.ppi = ppi        
+        if folder[-1] == "/":
+            folder = folder[:-1]
+        if not os.path.exists(folder+"/pcap"):
+            log_message = "PcapDump: Creating directory to store results"
+            logging.debug(log_message)
+            os.makedirs(folder+"/pcap")
+        self.__fh = open(folder+"/pcap"+savefile, mode='wb')
```

Comment:
The way I read this, it assumes that the user must always want the files in a /pcap folder either in their CWD by default or in another path they pass. If a tool wants to use such a restriction, that seems fine, but I think that the output serialization functions should accept whatever output path (savefile parameter) that the user wants - as people may want to do other things than save to SOMETHING/pcap/SOMETHING.